.github/workflows/python-tests.yml

```diff
@@ -1,68 +1,39 @@
 name: Run unit tests
 
 on:
-  pull_request:
   push:
-    branches: [ main ]
-    tags:
+  pull_request:
   workflow_dispatch:
 
-jobs:
-  tests:
-    name: ${{ matrix.name }} (${{ matrix.os }}, ${{ matrix.toxenv }})
-    runs-on: ${{ matrix.os }}
-    if: "!(contains(github.event.head_commit.message, '[skip ci]') || contains(github.event.head_commit.message, '[ci skip]'))"
-    strategy:
-      fail-fast: false
-      matrix:
-        include:
-          - name: Python 3.9 with required dependencies
-            os: windows-latest
-            python-version: 3.9
-            toxenv: py39-test-numpy122
-
-          - name: Python 3.10 with required dependencies
-            os: ubuntu-latest
-            python-version: "3.10"
-            toxenv: py310-test-numpy124
-
-          - name: Python 3.11 with required dependencies
-            os: macos-latest
-            python-version: "3.11"
-            toxenv: py311-test-astropy53
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
-          - name: Python 3.12 with required dependencies
-            os: ubuntu-latest
-            python-version: "3.12"
-            toxenv: py312-test
-            coverage: true
+jobs:
 
-          - name: Code style checks
-            os: ubuntu-latest
-            python-version: "3.12"
-            toxenv: codestyle
+  tests:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
+    with:
+      envs: |
+        - windows: py39-test-numpy122
+        - linux: py310-test-numpy124
+        - macos: py311-test-astropy53
+        - linux: py312-test
+        - linux: codestyle
+      coverage: 'codecov'
 
-    steps:
-    - uses: actions/checkout@v4
-      with:
-        fetch-depth: 0
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: python -m pip install tox
-    - name: Run tests
-      if: "! matrix.coverage"
-      run: tox -v -e ${{ matrix.toxenv }}
-    - name: Run tests with coverage
-      if: "matrix.coverage"
-      run: |
-        pip install Cython extension-helpers numpy
-        COVERAGE=1 pip install -e .[test]
-        pytest --pyargs astroscrappy docs --cov astroscrappy
-    - name: Upload coverage to codecov
-      if: "matrix.coverage"
-      run: |
-        pip install codecov
-        codecov
+  publish:
+    needs: tests
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/publish.yml@v1
+    with:
+      test_extras: test
+      test_command: pytest -p no:warnings --pyargs astroscrappy
+      targets: |
+        - cp*-manylinux_x86_64
+        - cp*-manylinux_aarch64
+        - cp*-macosx_x86_64
+        - cp*-macosx_arm64
+        - cp*-win_amd64
```

Comment:
we have some target configuration in `pyproject.toml`, should we delete it to avoid confusion ?

Reply:
Good catch